src/protocols/secure_channel/PairingSession.cpp

```diff
@@ -57,6 +57,59 @@ CHIP_ERROR PairingSession::ActivateSecureSession(const Transport::PeerAddress &
     return CHIP_NO_ERROR;
 }
 
+void PairingSession::Finish()
+{
+    Transport::PeerAddress address = mExchangeCtxt->GetSessionHandle()->AsUnauthenticatedSession()->GetPeerAddress();
+
+    // Discard the exchange so that Clear() doesn't try closing it. The exchange will handle that.
```

Comment:
This is a bit confusing, because the Clear method of this class does not try closing anything....  What should this really be saying?  Maybe something like:
```suggestion
    // Discard the exchange so that our subclasses don't try closing it. The exchange will handle that.
```